test/jdk/javax/swing/JToolTip/5047379/bug5047379.java

```diff
@@ -0,0 +1,227 @@
+import java.awt.Robot;
+
+import java.util.HashMap;
+import java.util.Map;
+
+
+public class bug5047379 {
+    private static final long TIMEOUT =  20 * 1000;
+    static Robot testRobot;
+    static toolTipTest testObj;
+    static Map<String, String> lookAndFeelMaps = new HashMap<String, String>();
+
+    public static void main(String[] args) throws Exception {
+        initMap();
+
+        testObj = new toolTipTest();
+        testRobot = new Robot();
+        SwingUtilities.invokeLater(new Runnable() {
+
+            public void run() {
+                try {
+                    testObj.runTest();
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            }
+        });
+
+        for (Map.Entry<String, String> value : lookAndFeelMaps.entrySet()) {
+
+            if (value.getKey().equals("Nimbus")) {
+                continue;
+            }
+
+            testRobot.setAutoDelay(50);
+            testRobot.waitForIdle();
+            Point movePoint = testObj.getButtonPoint(testObj.b);
+            testRobot.mouseMove(movePoint.x, movePoint.y);
+            testRobot.waitForIdle();
+
+            long timeout = System.currentTimeMillis() + 9000;
+            while (!testObj.isTooltipAdded && (System.currentTimeMillis() < timeout)) {
+                try {
+                    Thread.sleep(500);
+                } catch (Exception e) {
+                }
+            }
+            testObj.handleToolTip(value.getKey(),value.getValue());
+
+            Thread.sleep(1000);
+
+        }
+    }
+
+    public static void initMap() {
+        String sLnF;
+        String sMapKey;
+        UIManager.LookAndFeelInfo[] lookAndFeel = UIManager.getInstalledLookAndFeels();
+        for (UIManager.LookAndFeelInfo look : lookAndFeel) {
+
+            sLnF = look.getClassName();
+            sMapKey = sLnF.substring(sLnF.lastIndexOf(".")+1);
+            sMapKey = sMapKey.replaceAll("LookAndFeel","");
+            sMapKey = sMapKey.trim();
+
+            lookAndFeelMaps.put(sMapKey, sLnF);
+        }
+    }
+};
+
+class toolTipTest{
+    volatile boolean isTooltipAdded;
+    JFrame frame;
+    JButton a;
+    JButton b;
+    JButton c;
+    JButton d;
+
+    void handleToolTip(String key, String Value) throws Exception {
+        UIManager.setLookAndFeel(Value);
+        SwingUtilities.updateComponentTreeUI(frame);
+
+        SwingUtilities.invokeAndWait(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    JToolTip tooltip = (JToolTip) Util.findSubComponent(
+                            JFrame.getFrames()[0], "JToolTip");
+
+                    BasicToolTipUI toolTipObj = null;
+
+                    switch (key) {
+                        case "Metal":
+                            toolTipObj = (MetalToolTipUI) MetalToolTipUI.createUI(tooltip);
+                            break;
+                        case "WindowsClassic":
```

Comment:
Since it is supposed to be run on all platforms, this L&F cannot be hardcoded as this will not be applicable/present in all platforms